# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/linux24/05-devfs-not-exp.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

Currently (at version 2.0.x) ROCK Linux required DevFS. It is highly
useful. Later version of ROCK Linux are planed to support other variants
(like old-style device nodes or udevfs).

  - Rene Rebe <rene@rocklinux.org>

--- linux-2.4.23/fs/Config.in.orig	2003-11-28 19:26:21.000000000 +0100
+++ linux-2.4.23/fs/Config.in	2003-12-09 00:55:11.000000000 +0100
@@ -72,13 +72,9 @@
 
 bool '/proc file system support' CONFIG_PROC_FS
 
-# For some reason devfs corrupts memory badly on x86-64. Disable it 
-# for now.
-if [ "$CONFIG_X86_64" != "y" ] ; then
-dep_bool '/dev file system support (EXPERIMENTAL)' CONFIG_DEVFS_FS $CONFIG_EXPERIMENTAL
+dep_bool '/dev file system support (Required for ROCK Linux)' CONFIG_DEVFS_FS $CONFIG_EXPERIMENTAL
 dep_bool '  Automatically mount at boot' CONFIG_DEVFS_MOUNT $CONFIG_DEVFS_FS
 dep_bool '  Debug devfs' CONFIG_DEVFS_DEBUG $CONFIG_DEVFS_FS
-fi
 
 # It compiles as a module for testing only.  It should not be used
 # as a module in general.  If we make this "tristate", a bunch of people
